### DISCUSSION TITLE
TypeError: not a function

### DISCUSSION THREAD

**Discussion by AlephBaa** [vfsfitvnm/frida-il2cpp-bridge#353] [Q&A]
My goal is to read the returned value and modify that.
below is my ts code:

```
const myAssem = Il2Cpp.Domain.assembly("myAssem");
const MyClass = myAssem.image.class("Foo.Bar.MyClass");
const MyFunction = MyClass.method<any>("MyFunction");
		
MyFunction.implementation = function (arg1: any, arg2: any, arg3: any): any {
     var ret = MyFunction.invoke(arg1, arg2, arg3);
     console.log(ret.toString());
     return ret;
}
```

this returns TypeError: not a function on line 6: `var ret = ...`

I couldn't figure it out what's the problem about.

**Comment by vfsfitvnm** [comment]
Hmm, I don't really know, I need at least a stacktrace to answer. You'd better use the *latest* version of `frida-compile`

**Reply by vfsfitvnm** [reply]
Does the following work?
```ts
const myAssem = Il2Cpp.Domain.assembly("myAssem");
const MyClass = myAssem.image.class("Foo.Bar.MyClass");
		
MyClass.method<any>("MyFunction").overload("System.String", "System.String", "Foo.Bar.ThisEnum").implementation = function (arg1: any, arg2: any, arg3: any): any {
     var ret = this.method<any>("MyFunction").overload("System.String", "System.String", "Foo.Bar.ThisEnum").invoke(arg1, arg2, arg3);
     console.log(ret.toString());
     return ret;
}
```

**Reply by vfsfitvnm** [reply]
Hmm unfortunately I cannot help you further, there's clearly something wrong in the codebase - but I can't reproduce.
This error should be quite easy to debug, though, that's why I need a stack trace

**Comment by berkedel** [comment]
What's the type of the return value? Does it throw an error when you just `console.log(ret)` instead of `console.log(ret.toString())`?
